### PR TITLE
created the endpoint for identityURL

### PR DESCRIPTION
**Pull request by vinayak-trivedi**
closes https://github.com/Real-Dev-Squad/website-backend/issues/463

### CODE REVIEW

File: controllers/users.js [hunk 1]
```diff
@@ -142,12 +142,26 @@ const postUserPicture = async (req, res) => {
     return res.boom.badImplementation('An internal server error occurred')
   }
 }
-
+const identityURL = async (req, res) => {
+  try {
+    const userId = req.userData.id
+    await userQuery.addOrUpdate(req.body, userId)
```
Comment (swarajpure) [inline controllers/users.js:148]: What should be the schema/structure for `req.body`?

Comment (harshith-venkatesh) [inline controllers/users.js:148]: What kind of validations will be done for URL?
Correct me if I am missing something here and sharing schema/ structure here would be helpful 👍🏻

Comment (vinayak-trivedi) [reply controllers/users.js:148]: {
   identityURL: "https://abc.com"
}
This is req.body

File: test/integration/users.test.js [hunk 2]
```diff
@@ -227,6 +227,25 @@ describe('Users', function () {
...
+    })
+  })
+  describe('PATCH /users/identityURL', function () {
+    it('Should update the identityURL', function (done) {
+      chai
+        .request(app)
+        .patch('/users/identityURL')
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send({
+          identityURL: 'identityURL'
```
Comment (swarajpure) [inline test/integration/users.test.js:241]: This shouldn't be allowed right? We should allow only valid links. Tomorrow someone might send `joker` here, is that a valid request body? Please see if we can use this: https://joi.dev/api/?v=17.5.0#stringurioptions

**Review by lakshayman — DISMISSED** [review]
Looks good to me

File: controllers/users.js [hunk 3]
```diff
@@ -142,12 +142,26 @@ const postUserPicture = async (req, res) => {
...
+    const { user } = await userQuery.fetchUser({ userId })
```
Comment (ankurnarkhede) [inline controllers/users.js:149]: Not getting an error from `userQuery.addOrUpdate` itself means the value has been updated. We dont need to get the values from DB again here.

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

**Review by ankur-lt — APPROVED** [review]

**Pull request merged**